### PR TITLE
Deprecate storage-type flag

### PR DESCRIPTION
**Pull request by jimmidyson**
This uses the same conventions as Kubernetes, not marking flag as deprecated via pflag.FlagSet because this hides the flag from help, but rather marking the usage message as DEPRECATED.

As discussed in #2201, this keeps the flag and chart values to not break existing deployments, but completely unwires it so it can be removed in a (near-)future release.

### CODE REVIEW

**Comment by MHBauer** [comment]
why do we want to show the flag in help? to act as notification that it is deprecated?

**Comment by jimmidyson** [comment]
I've refactored this following https://github.com/spf13/pflag/pull/163 to use the "proper" `MarkDeprecated` function but also ensuring that it's visible in usage - this makes it more visible that it will be unused and removed in the next few releases.

File: cmd/apiserver/app/server/options.go [hunk 1]
```diff
@@ -23,23 +23,20 @@ import (
...
-	"github.com/kubernetes-incubator/service-catalog/pkg/registry/servicecatalog/server"
 )
 
 const (
 	// Store generated SSL certificates in a place that won't collide with the
 	// k8s core API server.
 	certDirectory = "/var/run/kubernetes-service-catalog"
 
-	storageTypeFlagName = "storageType"
+	storageTypeFlagName = "storage-type"
```
Comment (MHBauer) [inline cmd/apiserver/app/server/options.go]: this doesn't change the flag, as far as I can tell. We just weren't using it before.

Comment (jimmidyson) [reply cmd/apiserver/app/server/options.go]: It wasn't being used before, but it is now - see https://github.com/kubernetes-incubator/service-catalog/pull/2266/files#diff-5dd3f709c02872b9414e257945813685R89 and following lines.

**Comment by jimmidyson** [comment]
@carolynvs Any chance of a review of this?

File: cmd/apiserver/app/server/options.go [hunk 2]
```diff
@@ -86,12 +83,15 @@ func NewServiceCatalogServerOptions() *ServiceCatalogServerOptions {
...
-		"etcd",
+	// storage-type flag is deprecated so let's mark it as so but keep it visible in usage
+	// to make it more obvious that it will be removed in the near future.
+	_ = flags.String(
+		storageTypeFlagName,
+		"",
 		"The type of backing storage this API server should use",
 	)
+	flags.MarkDeprecated(storageTypeFlagName, "The value of this flag is now unused and will be removed in the near future")
+	flags.Lookup(storageTypeFlagName).Hidden = false
```
Comment (carolynvs) [inline cmd/apiserver/app/server/options.go]: 👍

**Comment by jimmidyson** [comment]
@carolynvs Thanks for the review! Is there anyone else that can review so we could get this merged? Thanks!

**Comment by carolynvs** [comment]
/assign @jboyd01 

Jay, I'm hoping you have time to give this a lgtm? 😀

**Comment by jboyd01** [comment]
sorry this has taken so long to get merged @jimmidyson, we appreciate the contribution!
/lgtm

**Pull request merged**